frontend/components/buttons/RevealButton/RevealButton.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import Button from "components/buttons/Button";
+import TooltipWrapper from "components/TooltipWrapper";
+
+export interface IRevealButtonProps {
+  showBoolean: boolean;
+  baseClass: string;
+  hideString: string;
+  showString: string;
+  caratBefore?: boolean;
+  caratAfter?: boolean;
+  autofocus?: boolean;
+  disabled?: boolean;
+  tooltipHtml?: string;
+  onClick?:
+    | ((value?: any) => void)
+    | ((evt: React.MouseEvent<HTMLButtonElement>) => void);
+}
+
+const RevealButton = ({
+  showBoolean,
+  baseClass,
+  hideString,
+  showString,
+  caratBefore,
+  caratAfter,
+  autofocus,
+  disabled,
+  tooltipHtml,
+  onClick,
+}: IRevealButtonProps): JSX.Element => {
+  const classNameGenerator = () => {
+    if (caratBefore) {
+      return showBoolean ? "reveal upcaratbefore" : "reveal rightcaratbefore";
+    }
+    if (caratAfter) {
+      return showBoolean ? "reveal upcaratafter" : "reveal downcaratafter";
+    }
+  };
+
+  if (tooltipHtml) {
+    return (
+      <Button
+        variant="unstyled"
+        className={`reveal-button ${classNameGenerator()}`}
+        onClick={onClick}
+        autofocus={autofocus}
+        disabled={disabled}
+      >
+        <TooltipWrapper tipContent={tooltipHtml}>
+          {showBoolean ? hideString : showString}
+        </TooltipWrapper>
+      </Button>
+    );
+  }
+
+  return (
+    <Button
+      variant="unstyled"
+      className={`${classNameGenerator()}
+                    ${baseClass}__reveal-button`}
+      onClick={onClick}
+      autofocus={autofocus}
+      disabled={disabled}
+    >
+      {showBoolean ? hideString : showString}
+    </Button>
+  );
```

Comment:
Is it possible to merge these? They seem to only have one difference. I'm seeing this:

```suggestion
   return (
    <Button
      variant="unstyled"
      className={`${classNameGenerator()}
                    ${baseClass}__reveal-button`}
      onClick={onClick}
      autofocus={autofocus}
      disabled={disabled}
    >
    {tooltipHtml ? (
      <TooltipWrapper tipContent={tooltipHtml}>
          {showBoolean ? hideString : showString}
        </TooltipWrapper>
     ) : ({showBoolean ? hideString : showString})
    </Button>
  );
```

Reply:
Yes! thank you! 7 pm brain